docs/develop/go/temporal-nexus.mdx

```diff
@@ -0,0 +1,346 @@
+	--namespace <your-target-namespace> \
+	--region us-west-2 \
+	--ca-certificate-file 'path/to/your/ca.pem' \
+	--retention-days 1
+```
+
+Alternatively, you can create Namespace through the UI: [https://cloud.temporal.io/Namespaces](https://cloud.temporal.io/Namespaces)
+
+### Create a Nexus Endpoint to route requests from caller to handler
+
+To create a Nexus Endpoint you must have a Developer account role or higher, and have NamespaceAdmin permission on the `--target-namespace`.
+
+```
+tcld nexus endpoint create \
+  --name <my-nexus-endpoint-name> \
+  --target-task-queue my-handler-task-queue \
+  --target-namespace <my-target-namespace.account> \
+  --allow-namespace <my-caller-namespace.account> \
+  --description-file description.md
+```
+
+The `--allow-namespace` is used to build an Endpoint allowlist of caller Namespaces that can use the Nexus Endpoint, as described in Runtime Access Control.
+
+Alternatively, you can create a Nexus Endpoint through the UI: [https://cloud.temporal.io/nexus](https://cloud.temporal.io/nexus)
+
+### Run Workers Connected to Temporal Cloud
+
+[View the source code](https://github.com/temporalio/samples-go/tree/main/nexus) in the context of the rest of the application code.
+
+Run the handler Worker:
+
+```go
+cd handler
+
+go run ./worker \
+	-target-host <your-target-namespace.account>.tmprl.cloud:7233 \
+	-namespace <your-target-namespace.account> \
+	-client-cert 'path/to/your/ca.pem' \
+	-client-key 'path/to/your/ca.key'
+```
+
+Run the caller Worker:
+
+```go
+cd caller
+
+go run ./worker \
+	-target-host <your-caller-namespace.account>.tmprl.cloud:7233 \
+	-namespace <your-caller-namespace.account> \
+	-client-cert 'path/to/your/ca.pem' \
+	-client-key 'path/to/your/ca.key'
+```
+
+### Start a caller Workflow
+
+In order to start the caller Workflow, run the starter.
+
+```go
+cd caller
+
+go run ./starter \
+	-target-host <your-caller-namespace.account>.tmprl.cloud:7233 \
+	-namespace <your-caller-namespace.account> \
+	-client-cert 'path/to/your/ca.pem' \
+	-client-key 'path/to/your/ca.key'
+```
+
+This will result in:
+
+```
+2024/07/23 19:57:40 Workflow result: Nexus Echo 👋
+2024/07/23 19:57:40 Workflow result: ¡Hola! Nexus 👋
+```
+
+### Observability
+
+#### Web UI
+
+A synchronous Nexus Operation will surface in the caller Workflow as follows, with just `NexusOperationScheduled` and `NexusOperationCompleted` events in the caller’s Workflow history:
+
+![Observability in Temporal Cloud Sync](/img/nexus-go/go-sdk-observability-sync.png)
+
+An asynchronous Nexus Operation will surface in the caller Workflow as follows, with `NexusOperationScheduled`, `NexusOperationStarted`, and `NexusOperationCompleted`, in the caller’s Workflow history:
+
+![Observability in Temporal Cloud Async](/img/nexus-go/go-sdk-observability-async.png)
+
+#### Temporal CLI
+
+Nexus pending operations and callbacks are included in the Workflow description:
+
+```
+temporal workflow describe -w <ID>
+```
```

Comment:
I would separate this to caller and handler workflows. Handler will have a callback attached while caller will see a pending nexus operation as long as the operation is pending completion.